File: app/src/main/java/com/example/gestionturnosapp/data/Turno.kt
package com.example.gestionturnosapp.data

import androidx.room.Entity
import androidx.room.PrimaryKey
import com.google.gson.annotations.SerializedName

@Entity(tableName = "turnos")
data class Turno(
    @PrimaryKey
    @SerializedName("_id", alternate = ["id"])
    val id: String,
    val pacienteNombre: String,
    val fecha: String,
    val hora: String,
    val motivo: String,
    val estado: String = "Pendiente",
    val especialidad: String? = "General",
    val doctor: String? = "Dr. Asignado"
)

File: backend/models/Turno.js
const mongoose = require('mongoose');

const TurnoSchema = new mongoose.Schema({
    usuarioId: { type: mongoose.Schema.Types.ObjectId, ref: 'User', required: true },
    pacienteNombre: { type: String, required: true },
    fecha: { type: String, required: true },
    hora: { type: String, required: true },
    motivo: { type: String, default: "General" },
    especialidad: { type: String, default: "General" },
    doctor: { type: String, default: "Dr. Asignado" },
    estado: { type: String, default: "Pendiente" }
});

module.exports = mongoose.model('Turno', TurnoSchema);
